src/acquisition/covidcast/database.py

```diff
@@ -232,52 +217,37 @@ def insert_or_update_batch(self, cc_rows, batch_size=2**20, commit_partial=False
 
   def run_dbjobs(self):
 
-    signal_load_set_comp_keys = f'''
-        UPDATE `{self.load_table}`
-        SET compressed_signal_key = md5(CONCAT(`source`,`signal`)),
-            compressed_geo_key = md5(CONCAT(`geo_type`,`geo_value`))
-    '''
-
-    signal_load_mark_batch = f'''
-        UPDATE `{self.load_table}` 
-        SET process_status = '{PROCESS_STATUS.BATCHING}'
-    '''
-
+    # we do this LEFT JOIN trick because mysql cant do set difference (aka EXCEPT or MINUS)
+    # (as in " select distinct source, signal from signal_dim minus select distinct source, signal from signal_load ")
     signal_dim_add_new_load = f'''
-        INSERT INTO signal_dim (`source`, `signal`, `compressed_signal_key`) 
-            SELECT DISTINCT `source`, `signal`, compressed_signal_key 
-                FROM `{self.load_table}` 
-                WHERE compressed_signal_key NOT IN 
-                    (SELECT DISTINCT compressed_signal_key 
-                     FROM signal_dim)
+        INSERT INTO signal_dim (`source`, `signal`)
+            SELECT DISTINCT sl.source, sl.signal
+                FROM {self.load_table} AS sl LEFT JOIN signal_dim AS sd
```

Comment:
Should we be using with (nolock) on our tables here? Or are we sure that no process will ever have the tables locked?

https://www.sqlshack.com/understanding-impact-clr-strict-security-configuration-setting-sql-server-2017/

Reply:
question stands for all queries in this file.

Reply:
in practice, locks from these statements should only last a very short time, on the order of (or less) than the time it takes to process one csv file (which is quite fast).  using this can also lead to inconsistent data reads for outside transactions, so its not worth any small reductions in time waiting for locks to be released.  (also, the article you referred to is for MS SQL Server though an equivalent exists for mysql)

Reply:
Gotcha, makes sense!